src/Type/Php/IsSubclassOfFunctionTypeSpecifyingExtension.php

```diff
@@ -42,6 +43,11 @@ public function specifyTypes(FunctionReflection $functionReflection, FuncCall $n
 		$allowStringType = isset($node->getArgs()[2]) ? $scope->getType($node->getArgs()[2]->value) : new ConstantBooleanType(true);
 		$allowString = !$allowStringType->equals(new ConstantBooleanType(false));
 
+		// prevent false-positives in IsAFunctionTypeSpecifyingHelper
+		if ($objectOrClassType instanceof GenericClassStringType && $classType instanceof GenericClassStringType) {
```

Comment:
obviously this is not a great fix, but with this PR we fix the existing false-positives.
proper support for generic-class-string in is_subclass_of could be added later.. since there are not existing tests for this case, this should be fine for now